src/libraries/Common/src/System/Reflection/Metadata/AssemblyNameInfo.cs

```diff
@@ -0,0 +1,222 @@
+            assemblyName.ContentType = ExtractAssemblyContentType(_flags);
+#pragma warning disable SYSLIB0037 // Type or member is obsolete
+            assemblyName.ProcessorArchitecture = ExtractProcessorArchitecture(_flags);
+#pragma warning restore SYSLIB0037 // Type or member is obsolete
+
+#if SYSTEM_PRIVATE_CORELIB
+            if (PublicKeyOrToken is not null)
+            {
+                if ((Flags & AssemblyNameFlags.PublicKey) != 0)
+                {
+                    assemblyName.SetPublicKey(PublicKeyOrToken);
+                }
+                else
+                {
+                    assemblyName.SetPublicKeyToken(PublicKeyOrToken);
+                }
+            }
+#else
+            if (!PublicKeyOrToken.IsDefault)
+            {
+#pragma warning disable RS0030 // TypeSystem does not allow System.Linq, but we need to use System.Linq.ImmutableArrayExtensions.ToArray
+                // A copy of the array needs to be created, as AssemblyName allows for the mutation of provided array.
+                if ((Flags & AssemblyNameFlags.PublicKey) != 0)
+                {
+                    assemblyName.SetPublicKey(PublicKeyOrToken.ToArray());
+                }
+                else
+                {
+                    assemblyName.SetPublicKeyToken(PublicKeyOrToken.ToArray());
+                }
+#pragma warning restore RS0030
+            }
+#endif
+
+            return assemblyName;
+        }
+
+        /// <summary>
+        /// Parses a span of characters into a assembly name.
+        /// </summary>
+        /// <param name="assemblyName">A span containing the characters representing the assembly name to parse.</param>
+        /// <returns>Parsed type name.</returns>
+        /// <exception cref="ArgumentException">Provided assembly name was invalid.</exception>
+        public static AssemblyNameInfo Parse(ReadOnlySpan<char> assemblyName)
+            => TryParse(assemblyName, out AssemblyNameInfo? result)
+                ? result!
+#if SYSTEM_REFLECTION_METADATA || SYSTEM_PRIVATE_CORELIB
+                : throw new ArgumentException(SR.InvalidAssemblyName, nameof(assemblyName));
+#else // tools that reference this file as a link
+                : throw new ArgumentException("The given assembly name was invalid.", nameof(assemblyName));
+#endif
+
+        /// <summary>
+        /// Tries to parse a span of characters into an assembly name.
+        /// </summary>
+        /// <param name="assemblyName">A span containing the characters representing the assembly name to parse.</param>
+        /// <param name="result">Contains the result when parsing succeeds.</param>
+        /// <returns>true if assembly name was converted successfully, otherwise, false.</returns>
+        public static bool TryParse(ReadOnlySpan<char> assemblyName,
+#if SYSTEM_REFLECTION_METADATA || SYSTEM_PRIVATE_CORELIB // required by some tools that include this file but don't include the attribute
```

Comment:
If would be better to make the tools project include the polyfill attribute instead of ifdefing. It is out standard way to with these type of situations.

Reply:
> If would be better to make the tools project include the polyfill attribute instead of ifdefing. It is out standard way to with these type of situations.

I was hoping to avoid that by updating the tools ASAP to use latest SDK, remove the code links and just consume public APIs. Is that acceptable?